packages/fsengagement/src/EngagementService.ts

```diff
@@ -237,6 +241,57 @@ export class EngagementService {
       });
   }
 
+  async getInboxMessages(attributes?: AttributePayload): Promise<EngagementMessage[]> {
+    // check we have a user profile
+    if (!this.profileId) {
+      throw new Error('Profile not loaded.');
+    }
+
+    // cache
+    if (this.messages.length) {
+      if (+new Date() - this.messageCache < this.cacheTTL) {
+
+        return Promise.resolve(this.messages);
+      }
+    }
+
+    const lastEngagementFetch = await AsyncStorage.getItem('LAST_ENGAGEMENT_FETCH');
+    return this.networkClient.post(`/PublishedMessages/getInboxForProfile/${this.profileId}`,
+      JSON.stringify(attributes))
+      .then((r: any) => r.data)
+      .then((list: any) => list.map((data: any) => {
+        return {
+          id: data.id,
+          published: new Date(data.published),
+          isNew: lastEngagementFetch ?
+            moment.utc(data.published).valueOf() > parseInt(lastEngagementFetch, 10) : false,
+          message: JSON.parse(data.message),
+          title: data.title,
+          inbox: data.inbox,
+          attributes: data.attributes
+        };
+      }))
+      .then((messages: EngagementMessage[]) => {
+        this.messages = messages;
+        this.messageCache = +new Date();
+        AsyncStorage.setItem('LAST_ENGAGEMENT_FETCH', moment.utc(new Date()).valueOf().toString())
```

Comment:
Because moment is such a large library, could you switch to using just the base Date functions or a smaller library to format this?

Reply:
Sure, this code is already merged into the rnn2 branch, should we change it there as well?

Reply:
Yeah, I noticed moment was added there when I was doing a merge, but wasn't sure what it was for. If you could make the same change there, I would appreciate it.

Reply:
Sorry, I forgot that we should be able to just merge develop into that branch and it should make the same change. I hope you haven't put a bunch of work into that.